### PR TITLE
Convert to addon v2

### PR DESCRIPTION
**Pull request by AnastasiiaPlutalova**

### CODE REVIEW

File: addon/rollup.config.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+  plugins: [
+    // These are the modules that users should be able to import from your
+    // addon. Anything not listed here may get optimized away.
+    addon.publicEntrypoints(['initializers/**/*.js', 'utils/**/*.js']),
+
+    // These are the modules that should get reexported into the traditional
+    // "app" tree. Things in here should also be in publicEntrypoints above, but
+    // not everything in publicEntrypoints necessarily needs to go here.
+    addon.appReexports(['initializers/**/*.js', 'utils/**/*.js']),
```
Comment (SergeAstapov) [inline addon/rollup.config.js]: @AnastasiiaPlutalova this should be

```js
addon.appReexports(['initializers/**/*.js', 'services/**/*.js'])
```

File: addon/rollup.config.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+export default {
+  // This provides defaults that work well alongside `publicEntrypoints` below.
+  // You can augment this if you need to.
+  output: addon.output(),
+
+  plugins: [
+    // These are the modules that users should be able to import from your
+    // addon. Anything not listed here may get optimized away.
+    addon.publicEntrypoints(['initializers/**/*.js', 'utils/**/*.js']),
```
Comment (SergeAstapov) [inline addon/rollup.config.js]: @AnastasiiaPlutalova I think this should be

```js
addon.appReexports(['initializers/**/*.js', 'services/**/*.js', 'utils/**/*.js'])
```

**Review by SergeAstapov — CHANGES_REQUESTED** [review]

**Comment by SergeAstapov** [comment]
@AnastasiiaPlutalova @villander as I follow up, I suggest to add `release-it` setup like it was done in https://github.com/adopted-ember-addons/ember-keyboard/pull/597/files

File: addon/package.json [hunk 3]
```diff
@@ -27,11 +39,15 @@
     "lint:js:fix": "eslint . --fix"
   },
   "dependencies": {
-    "ember-cli-babel": "^7.26.10"
```
Comment (villander) [inline addon/package.json:30]: we don't need it anymore?

File: addon/package.json [hunk 4]
```diff
@@ -57,5 +74,19 @@
...
+    "type": "addon",
+    "version": 2,
+    "app-js": {
+      "./initializers/ember-engines-router-service.js": "./dist/_app_/initializers/ember-engines-router-service.js",
+      "./utils/namespace-engine-route-name.js": "./dist/_app_/utils/namespace-engine-route-name.js",
+      "./utils/resembles-url.js": "./dist/_app_/utils/resembles-url.js",
+      "./utils/root-owner.js": "./dist/_app_/utils/root-owner.js"
+    }
+  },
+  "volta": {
```
Comment (villander) [inline addon/package.json]: I don't want to use volta, I want to keep this addon agnostic from Node version management ; )

File: addon/rollup.config.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    // `dependencies` and `peerDependencies` as well as standard Ember-provided
+    // package names.
+    addon.dependencies(),
+
+    // Ensure that standalone .hbs files are properly integrated as Javascript.
+    addon.hbs(),
+
+    // addons are allowed to contain imports of .css files, which we want rollup
+    // to leave alone and keep in the published output.
+    addon.keepAssets(['**/*.css']),
```
Comment (villander) [inline addon/rollup.config.js:48]: We're not running CSS, why do we need it?

**Review by villander — CHANGES_REQUESTED** [review]
I left some important suggestions. Also, CI is broken with this new format.

File: addon/rollup.config.js (hunk 5 above)
Comment (SergeAstapov) [reply addon/rollup.config.js:48]: @villander this comes from blueprint so I would keep it for the sake of alignment with blueprint

Comment (villander) [reply addon/rollup.config.js:48]: Ok, sounds good to me.

File: addon/package.json (hunk 3 above)
Comment (AnastasiiaPlutalova) [reply addon/package.json:30]: No, according to tutorial part 4 step 3 https://github.com/embroider-build/embroider/blob/main/PORTING-ADDONS-TO-V2.md#part-4-convert-addon-to-v2

File: addon/package.json (hunk 4 above)
Comment (AnastasiiaPlutalova) [reply addon/package.json]: removed it

File: test-app/package.json [hunk 6]
```diff
@@ -90,5 +90,10 @@
   },
   "ember": {
     "edition": "octane"
+  },
+  "ember-addon": {
```
Comment (SergeAstapov) [inline test-app/package.json]: @AnastasiiaPlutalova I don't think we need this

**Review by SergeAstapov — APPROVED** [review]
This looks great now!

**Review by villander — APPROVED** [review]

**Pull request merged**